File: benchmarks/oopsla25/subjects/076.py
import sys

def f076(s: str):  
  # return s[:-2]
  return s[:len(s)-2]

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f076(my_string))

File: benchmarks/oopsla25/subjects/076.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

char* f076(const char *s) {
    size_t len = strlen(s);
    char* result = (char*)malloc(len - 1);
    if (result) {
        strncpy(result, s, len - 2);
        result[len - 2] = '\0';
    }    
    return result;
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  char *s = f076(my_string);
  printf("%s\n", s);
  return 0;
}
